src/runtime/moduleobject.cs

```diff
@@ -181,7 +187,23 @@ public void LoadNames()
                 {
                     continue;
                 }
-                GetAttribute(name, true);
+
+                if(GetAttribute(name, true) != null)
+                {
+                    // if it's a valid attribute, add it to __all__
+                    var pyname = Runtime.PyString_FromString(name);
+                    try
+                    {
+                        if (Runtime.PyList_Append(new BorrowedReference(__all__), pyname) != 0)
```

Comment:
Frankly, I screwed up `PyList_Append`. Its only overload schizophrenically mixes references and raw pointers. Can you change it to either have two overloads with only references and with only pointers, or just one that takes only references?

Reply:
There are only a few calls to `PyList_Append` throughout the code base, I changed all of them to use only references.